File: src/github/Disconnect.cs
using System;
using Robomotion;
using Octokit;

namespace Github
{
    [Color("#003b57"), Icon(Icons.mdiGithubCircle), Inputs(1), Outputs(1)]
    [NodeName("Robomotion.Github.Disconnect"), Title("Disconnect")]
    public class Disconnect : Node
    {


        [Default("Message", "conn_id"), MessageScope, Title("Connection Id"), Output]
        public InVariable<string> ConnectionId { get; set; }

        public override void OnCreate()
        {
            
        }

        public override void OnMessage(ref Context ctx)
        {
        GitHubClient github = null;

        string connId = ConnectionId.Get(ctx);
        if (string.IsNullOrEmpty(connId)) {
            throw new Error("ErrInvalidArg", "Connection id can not be empty");
        }

            Github.mutex.WaitOne();

        if (!Github.connections.TryGetValue(connId, out github))
        {
            Github.mutex.ReleaseMutex();
            throw new Error("ErrInvalidArg", "Connection id not found");
        }
        Github.connections.Remove(connId);
        
        }

        public override void OnClose()
        {

        }
    }
}

File: src/language/v1/disconnect.go
package language

import (
	"github.com/robomotionio/robomotion-go/message"
	"github.com/robomotionio/robomotion-go/runtime"
)

type Disconnect struct {
	runtime.Node `spec:"id=Robomotion.GoogleNaturalLanguage.Disconnect,name=Disconnect,icon=mdiLanDisconnect,color=#9C27B0"`

	//Input
	InClientID runtime.InVariable[string] `spec:"title=Client ID,type=string,scope=Message,name=client_id,messageScope,customScope"`

	//Output

	//Options
}

func (n *Disconnect) OnCreate() error {

	return nil
}

func (n *Disconnect) OnMessage(ctx message.Context) error {
	connection_id, err := n.InClientID.Get(ctx)
	if err != nil {
		return err
	}
	removeClient(connection_id)

	return nil

}

func (n *Disconnect) OnClose() error {

	return nil
}
